src/deluge/gui/menu_item/arpeggiator/midi_cv/note_probability.h

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2014-2023 Synthstrom Audible Limited
+ *
+ * This file is part of The Synthstrom Audible Deluge Firmware.
+ *
+ * The Synthstrom Audible Deluge Firmware is free software: you can redistribute it and/or modify it under the
+ * terms of the GNU General Public License as published by the Free Software Foundation,
+ * either version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program.
+ * If not, see <https://www.gnu.org/licenses/>.
+ */
+#pragma once
+#include "gui/menu_item/integer.h"
+#include "gui/ui/sound_editor.h"
+#include "model/clip/instrument_clip.h"
+#include "model/song/song.h"
+
+namespace deluge::gui::menu_item::arpeggiator::midi_cv {
+class NoteProbability final : public Integer {
+public:
+	using Integer::Integer;
+	void readCurrentValue() override {
+		auto* current_clip = getCurrentInstrumentClip();
+		int64_t value = (int64_t)current_clip->arpeggiatorNoteProbability;
+		this->setValue((value * kMaxMenuValue + 2147483648) >> 32);
```

Comment:
```suggestion
		this->setValue((value * kMaxMenuValue + std::numeric_limits<int32_t>::max() + 1) >> 32);
```

Reply:
better with this. I'll make this change:

		this->setValue(computeCurrentValueForArpMidiCvRatchetsOrRhythm(value));